src/provider/treeView.ts

```diff
@@ -103,13 +103,28 @@ export class MaestroWorkBenchTreeViewProvider implements vscode.TreeDataProvider
             return this.shouldIncludeFile(relativePath);
         });
 
-        const filePaths = filteredItems
+        const nonEmptyFolders = await Promise.all(
+            filteredItems
+                .filter(item => item.contextValue === FileType.Folder)
+                .map(async (folder) => {
+                    const folderPath = folder.resourceUri.fsPath;
+                    const hasMatchingFiles = await this.hasMatchingFiles(folderPath);
+                    return hasMatchingFiles ? folder : null;
```

Comment:
Can't this just be another filter?

Reply:
Updatd in cc08df125e33343672f670fd118b679a0298749f